File: TDDBC_VendorMachine/Posaune/src/VendingMachine/VendingMachine/Money.cs
using System;

namespace VendingMachine
{
    public class Money
    {
        public Money(MoneyKind kind)
        {
            Kind = kind;

            switch (kind)
            {
                case MoneyKind.Yen1:
                    Amount = 1;
                    break;
                case MoneyKind.Yen5:
                    Amount = 5;
                    break;
                case MoneyKind.Yen10:
                    Amount = 10;
                    break;
                case MoneyKind.Yen50:
                    Amount = 50;
                    break;
                case MoneyKind.Yen100:
                    Amount = 100;
                    break;
                case MoneyKind.Yen500:
                    Amount = 500;
                    break;
                case MoneyKind.Yen1000:
                    Amount = 1000;
                    break;
                case MoneyKind.Yen5000:
                    Amount = 5000;
                    break;
                case MoneyKind.Yen10000:
                    Amount = 10000;
                    break;
                default:
                    throw new ArgumentOutOfRangeException("kind");
            }
        }

        public MoneyKind Kind { get; private set; }

        public int Amount { get; private set; }
    }
}

File: TDDBC_VendorMachine/HappyLuckyAkira/VendingMachine/Product/Money.cpp
/*
 * Money.cpp
 *
 *  Created on: 2012/09/30
 *      Author: Akira
 */

#include "Money.h"

Money::~Money()
{

}
Money::Money(int money)
  :val_(money)
{

}
bool Money::operator==(const Money& money)
{
  if (this->val_ == money.val_){ return true ;}
  return false;

}

//使えるお金かを確認する。
bool IsValidMoney(Money money)
{
  if (money == Money::Yen10()   ||
      money == Money::Yen100()  ||
      money == Money::Yen500()  ||
      money == Money::Yen1000()){
      return true;
  }
  return false;
}
